### ISSUE TITLE
Format on save issue

### ISSUE THREAD

**Issue by Meshredded** [Meshredded/vscode-robusta#9]
**Current behaviour :** 

The .jvs code is getting formatted on save regardless of the value of _robusta.formatOnSave_ property.

**Expected behaviour:**
- The .jvs code should not get formatted if the property robusta.formatOnSave is set to false.
- If _robusta.formatOnSave_ is set to true and _robusta.compileOnSave_ is also set to true : make sure to let finish the formatting task before running the compiling task.

**Comment by manekinekko** [comment]
I am working on this!

**Comment by Meshredded** [comment]
You rock man !

**Comment by manekinekko** [comment]
After investigating this, it seems that [your code](https://github.com/Meshredded/vscode-robusta/blob/master/src/runner/runner.ts#L30-L38) is working as expected. 

I tried your extension locally, and I can format (or not) the `.jvs` file based on the value of `robusta.formatOnSave`. Same thing for compiling.

Did I miss something?

**Comment by Meshredded** [comment]
> After investigating this, it seems that [your code](https://github.com/Meshredded/vscode-robusta/blob/master/src/runner/runner.ts#L30-L38) is working as expected.
> 
> I tried your extension locally, and I can format (or not) the `.jvs` file based on the value of `robusta.formatOnSave`. Same thing for compiling.
> 
> Did I miss something?

I think the problem occurs when the property `editor.formatOnSave` is set to `true`, in this case the `RobustaFormatingProvider` is called even if  `robusta.formatOnSave` is set to `false`. because `RobustaFormatingProvider` is [registered](https://github.com/Meshredded/vscode-robusta/blob/ad9dfc4083c2f2369f0b9159984e435104e6aca4/src/extension.ts#L7) as a formatting provider to vscode.

**Comment by manekinekko** [comment]
Oh! Gotcha! I can reproduce the issue when setting `editor.formatOnSave` to `true`. I will investigate this issue.

**Comment by manekinekko** [comment]
I did some researches and it seems that the best options is to remove the `robusta.formatOnSave` option and use `editor.formatOnSave ` like so:

```
// Set the default
"editor.formatOnSave": false,

// Enable per-language
"[jvs]": {
    "editor.formatOnSave": true
}
```

Also, your current implementation of the formatter is the correct approach, according to the [documentation](https://code.visualstudio.com/blogs/2016/11/15/formatters-best-practices#_the-formatting-api).

**Comment by Meshredded** [comment]
> I did some researches and it seems that the best options is to remove the `robusta.formatOnSave` option and use `editor.formatOnSave ` like so:
> 
> ```
> // Set the default
> "editor.formatOnSave": false,
> 
> // Enable per-language
> "[jvs]": {
>     "editor.formatOnSave": true
> }
> ```
> 
> Also, your current implementation of the formatter is the correct approach, according to the [documentation](https://code.visualstudio.com/blogs/2016/11/15/formatters-best-practices#_the-formatting-api).

You're right, that seems to be the issue here, I think we should move the `robusta.formatOnSave` and the [corresponding action](https://github.com/Meshredded/vscode-robusta/blob/335275ab4456629f38f0c578e34882358724815f/src/runner/runner.ts#L34) as well, to let the formatting provider do the the job by it self.
